.github/workflows/ci.yaml

```diff
@@ -17,42 +17,74 @@ on:
 
 jobs:
   test:
-    name: Test on Python ${{ matrix.python-version }} and Airflow ${{ matrix.airflow-version }}
+    name: Test on Python ${{ matrix.python-version }} and Airflow ${{ matrix.airflow-version }} and dbt ${{ matrix.dbt-version }}
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9, '3.10', '3.11']
-        airflow-version: ['2.5.1', '2.4.3']
+        python-version:
+          - '3.11'
+          - '3.10'
+          - '3.9'
+          - '3.8'
+        airflow-version:
+          - '2.7.2'
+          - '2.6.3'
+          - '2.5.3'
+          - '2.4.3'
+        dbt-version:
+          - 1.7
+          - 1.6
+          - 1.5
+          - 1.4
         exclude:
-          # No constraints available
-          - python-version: '3.11'
+          # Incompatible combinations
+          - python-version: 3.11
             airflow-version: '2.4.3'
 
+          - python-version: 3.11
+            airflow-version: '2.5.3'
+
     runs-on: ubuntu-latest
     steps:
       - name: Harden Runner
-        uses: step-security/harden-runner@v1
+        uses: step-security/harden-runner@v2.6.1
         with:
           egress-policy: block
           allowed-endpoints: >
             api.github.com:443
             files.pythonhosted.org:443
             hub.getdbt.com:443
+            github.com:80
             github.com:443
+            gitlab.com:80
+            gitlab.com:443
             objects.githubusercontent.com:443
             raw.githubusercontent.com:443
             pypi.org:443
-
-      - uses: actions/checkout@v3
+            archive.ubuntu.com:80
+            azure.archive.ubuntu.com:80
+            esm.ubuntu.com:443
+            motd.ubuntu.com:443
+            packages.microsoft.com:80
+            ppa.launchpadcontent.net:443
+            security.ubuntu.com:80
+
+      - run: |
+          sudo apt-get update
+          sudo apt-get install --yes --no-install-recommends postgresql
+
+      - uses: actions/checkout@v4.1.1
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v4.7.1
         with:
           python-version: ${{ matrix.python-version }}
+          architecture: x64
+          token: ${{ secrets.GH_READ_TOKEN }}
```

Comment:
Not clear to me why we need these two new arguments. The [docs](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-setup-python-on-ghes) for `setup-python` mention token only when discussing usage with GHES, which to be honest it's the first time I am learning about that.

Reply:
And architecture x64 is the default: https://github.com/actions/setup-python/blob/main/README.md?plain=1#L49

Reply:
I'll drop these two and see what happens, if CI is green I'll ship this.

Reply:
These two arguments were necessary for local testing with `act`. I think they can be removed

Reply:
Looks like we are green. Let's release.